tests/cephfs/cephfs_scale/cephfs_scale_utils.py

```diff
@@ -0,0 +1,548 @@
+        mds_logging_thread = threading.Thread(
+            target=self.run_mds_commands_periodically, args=(ceph_cluster, log_dir)
+        )
+        mds_logging_thread.daemon = (
+            True  # Make the thread a daemon thread to exit when the main program exits
+        )
+        mds_logging_thread.start()
+
+    def stop_mds_logging(self):
+        """
+        Stop the periodic logging of ceph-mds top output.
+        """
+        global stop_event
+        stop_event.set()  # Set the stop event to stop the logging thread
+        if mds_logging_thread is not None:
+            mds_logging_thread.join()
+
+    def run_commands_periodically(
+        self,
+        client,
+        cmd_list,
+        log_dir,
+        interval=60,
+    ):
+        """
+        Run the provided commands periodically.
+
+        :param client: Client object to execute the commands
+        :param cmd_list: List of commands to execute
+        :param interval: Time interval in seconds between each run
+        """
+
+        while not stop_event.is_set():
+            for cmd in cmd_list:
+                self.collect_ceph_details(client, cmd, log_dir)
+            stop_event.wait(interval)
+
+    def start_logging(self, client, cmd_list, log_dir):
+        """
+        Start the logging of Ceph commands periodically.
+
+        :param client: Client object to execute the commands
+        :param cmd_list: List of commands to execute
+        """
+        global logging_thread, stop_event
+        stop_event.clear()  # Clear the stop event before starting the thread
+        logging_thread = threading.Thread(
+            target=self.run_commands_periodically, args=(client, cmd_list, log_dir)
+        )
+        logging_thread.daemon = (
+            True  # Make the thread a daemon thread to exit when the main program exits
+        )
+        logging_thread.start()
+
+    def stop_logging(self):
+        """
+        Stop the periodic logging of Ceph commands.
+        """
+        global stop_event
+        stop_event.set()  # Set the stop event to stop the logging thread
+        if logging_thread is not None:
+            logging_thread.join()
+
+    def run_io_operations_parallel(
+        self,
+        mount_paths,
+        io_type,
+        io_operation=None,
+        io_threads=None,
+        io_file_size=None,
+        io_files=None,
+    ):
+        """
+        Run IO operations on the mounted paths in parallel based on the specified IO type.
+
+        :param mount_paths: List of dictionaries containing client and mount path
+        :param io_type: Type of IO operation ("smallfile" or "largefile")
+        :param io_operation: Operation type for smallfile (e.g., create, read, etc.)
+        :param io_threads: Number of threads for smallfile
+        :param io_file_size: File size for smallfile
+        :param io_files: Number of files for smallfile
+        """
+
+        def run_smallfile_io(client, mount_path):
```

Comment:
Can we use other existing IO tools and cmds for IO like - dd, crefi, tar and untar cmds ?

Reply:
Yes, I am including fio for now. other will be added later during further enhancements